### PR TITLE
Harden privileged capability repair and catalog approval boundaries

### LINKED ISSUE #248: Batch H: privileged autonomy boundary hardening
## Summary
Narrow Seraph's privileged autonomy boundaries so broader workflow leverage does not blur the line between planning, approval, execution, and credential use.

## Why
As Seraph becomes more autonomous, the next failure mode is hidden privileged capability bleed across delegated work, workflow resume paths, connector-backed actions, and secret-bearing execution. The current trust model is much better than before, but it still needs a stricter planner-versus-executor split and clearer credential-bound action envelopes.

## Internal PR-sized slices
- [x] planner-executor boundary separation v1
  ensure planning paths cannot directly inherit privileged execution capabilities that should stay behind explicit execution surfaces
- [x] credential-bound action envelopes v1
  route connector and vault-backed actions through narrower audited envelopes instead of loosely shared execution contexts
- [x] approval, replay, and resume revalidation v2
  extend fail-closed boundary checks across more autonomous replay, redirect, and delegated execution paths
- [x] privileged-path evals and abuse proofs v2
  add deterministic tests for boundary bypass attempts, stale-approval reuse, and credential misuse or recovery seams

## Done when
- planning paths are clearly separated from privileged execution paths
- credential-backed actions are explicit, narrow, and auditable
- approval, replay, and resume semantics fail closed when the privileged surface changes
- deterministic tests or evals cover abuse and recovery seams

## Workflow note
- Aggregate PR: #252
- Internal slices stay in this issue checklist unless a slice is intentionally split into its own standalone PR.

## Docs to update
- docs/implementation/STATUS.md
- docs/implementation/00-master-roadmap.md
- docs/implementation/01-trust-boundaries.md
- docs/implementation/02-execution-plane.md
- docs/implementation/10-superiority-delivery.md

### PR DESCRIPTION
**Pull request by natgurlain**
Closes #248

## Done on develop
- [x] workflow approvals and workflow replay/resume already bind to explicit approval-context snapshots instead of reusing stale privileged-surface assumptions
- [x] capability bootstrap already separates low-risk autorepair from manual repair surfaces instead of silently mutating policy or capability reach from planning alone
- [x] direct extension lifecycle actions already route through approval when the validated package requires it

## Working in this PR
- [x] reuse one risk-based lifecycle-approval seam for direct catalog installs and starter-pack install paths before capability expansion starts
- [x] keep low-risk bundled skill installs direct while requiring approval for bundled MCP-backed capability expansion that carries lifecycle-risk boundaries
- [x] gate generated multi-step privileged repair bundles in the cockpit so one click cannot chain multiple privilege-changing mutations
- [x] keep single explicit privileged repair actions available, update eval/contracts, and record the Batch H review findings in the implementation docs

## Still to do after this PR
- [ ] deepen planner-versus-executor separation beyond capability repair/install paths
- [ ] extend fail-closed privilege revalidation across more delegated and future workflow continuation paths
- [ ] add denser operator affordances for step-by-step privileged repairs

## Validation
- `python3 -m py_compile backend/src/api/capabilities.py backend/src/api/catalog.py backend/tests/test_capabilities_api.py backend/tests/test_catalog_api.py backend/tests/test_eval_harness.py`
- `cd backend && .venv/bin/python -m pytest tests/test_capabilities_api.py -q`
- `cd backend && .venv/bin/python -m pytest tests/test_catalog_api.py -q`
- `cd backend && .venv/bin/python -m pytest tests/test_eval_harness.py::test_runtime_eval_scenarios_expose_expected_details tests/test_eval_harness.py::test_run_runtime_evals_passes_all_scenarios -q`
- `cd frontend && NODE_OPTIONS=--experimental-require-module npm test -- --run src/components/cockpit/CockpitView.test.tsx`
- `cd frontend && npm run build`
- `cd docs && npm run build`
- `git diff --check`

### CODE REVIEW

File: backend/src/api/capabilities.py [hunk 1]
```diff
@@ -781,7 +777,11 @@ async def _activate_starter_pack_by_name(name: str) -> dict[str, Any]:
     installed_catalog_items: list[dict[str, Any]] = []
     missing_entries: list[str] = []
 
-    for item_name in [str(item) for item in pack.get("install_items", [])]:
+    install_item_names = [str(item) for item in pack.get("install_items", [])]
+    for item_name in install_item_names:
+        await require_catalog_install_approval(item_name)
+
```
[the remark this answers is not in the document]
Comment (natgurlain) [reply backend/src/api/capabilities.py:783]: Fixed in `ad5c630`.

Root cause: starter-pack activation was preflighting every `install_item` through the lifecycle approval seam, but that preflight was consuming approved fingerprints before any install mutation happened. In a multi-item privileged pack, item A could consume its approval, item B could still return `approval_required`, and the retry would force item A to be re-approved.

Resolution:
- added a non-consuming approved check in the approval repository
- made lifecycle approval support `consume=False` for preflight
- starter-pack activation now preflights all install items with `consume=False`, then consumes approval only when each item is actually installed
- added regressions in `test_catalog_api.py` and `test_capabilities_api.py` to pin the non-consuming preflight and the per-item consume sequence

Validation rerun locally:
- `cd backend && .venv/bin/python -m pytest tests/test_capabilities_api.py tests/test_catalog_api.py -q` -> `62 passed`
- `cd docs && npm run build`
- `git diff --check`
